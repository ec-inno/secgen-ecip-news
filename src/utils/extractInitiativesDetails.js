const extractInitiativesDetails = ({ details, locale }) => {
  const linguisticVersions =
    details && Object.keys(details).length !== 0
      ? Object.values(details.linguisticVersions)
      : [];

  // Try to get content for the current locale.
  let linguisticVersionIsFallback = false;
  let linguisticVersion = linguisticVersions.find(
    version => version.languageCode.toLowerCase() === locale
  );

  if (!linguisticVersion) {
    // Fallback to original language.
    // Information display should be "all or nothing", so we override, and all other props are coming from the original.
    linguisticVersion = linguisticVersions.find(version => version.original);
    linguisticVersionIsFallback = true;
  }

  /**
   * Information from `details`.
   */
  const status = details && details.status ? details.status : '';
  const deadline = details && details.deadline ? details.deadline : '';
  const registrationNumber =
    details && details.comRegNum ? details.comRegNum : '';
  const progress = details && details.progress ? details.progress : [];
  const funding = details && details.funding ? details.funding : {};
  const members = details && details.members ? details.members : [];
  const submission = details && details.submission ? details.submission : {};
  const isPartiallyRegistered = details && details.partiallyRegistered;
  const dateRefusal = details.refusalDate ? details.refusalDate : '';
  const dateRegistration = details.registrationDate
    ? details.registrationDate
    : '';
  const supportLink = details.supportLink ? details.supportLink : '';
  const refusalReasons = details.refusalReasons ? details.refusalReasons : [];
  const dateStart =
    details && details.startCollectionDate ? details.startCollectionDate : '';
  const dateEnd =
    details && details.earlyClosureDate ? details.earlyClosureDate : '';

  /**
   * Information from `linguisticVersion`.
   */
  const title =
    linguisticVersion && linguisticVersion.title
      ? linguisticVersion.title
      : '...';

  const draftLegal =
    linguisticVersion && linguisticVersion.draftLegal
      ? linguisticVersion.draftLegal
      : {};

  const additionalDocument =
    linguisticVersion && linguisticVersion.additionalDocument
      ? linguisticVersion.additionalDocument
      : {};

  const annexText =
    linguisticVersion && linguisticVersion.annexText
      ? linguisticVersion.annexText
      : '';

  const treaties =
    linguisticVersion && linguisticVersion.treaties
      ? linguisticVersion.treaties
      : '';

  const website =
    linguisticVersion && linguisticVersion.website
      ? linguisticVersion.website
      : '';

  const objectives =
    linguisticVersion && linguisticVersion.objectives
      ? linguisticVersion.objectives
      : '';

  const decisionUrl =
    linguisticVersion && linguisticVersion.decisionUrl
      ? linguisticVersion.decisionUrl
      : '';

  return {
    additionalDocument,
    annexText,
    dateEnd,
    dateRefusal,
    dateRegistration,
    dateStart,
    deadline,
    decisionUrl,
    draftLegal,
    funding,
    isPartiallyRegistered,
    linguisticVersionIsFallback,
    members,
    objectives,
    progress,
    refusalReasons,
    registrationNumber,
    status,
    submission,
    supportLink,
    title,
    treaties,
    website,
  };
};

export default extractInitiativesDetails;
