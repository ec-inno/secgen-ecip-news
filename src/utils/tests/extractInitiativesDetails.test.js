import extractInitiativesDetails from '../extractInitiativesDetails';

import ongoingInitiative from './stubs/initiatives/ongoing';

describe('extractInitiativesDetails utlity', () => {
  it('ongoing initiative case: English locale', () => {
    const result = extractInitiativesDetails({
      details: ongoingInitiative,
      locale: 'en',
    });
    expect(result).toMatchSnapshot();
  });
});
