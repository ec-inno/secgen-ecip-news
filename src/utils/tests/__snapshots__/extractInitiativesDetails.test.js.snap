// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractInitiativesDetails utlity ongoing initiative case: English locale 1`] = `
Object {
  "additionalDocument": Object {
    "id": 17003,
    "language": "en",
    "mimeType": "text/plain",
    "name": "testAnnex5.txt",
    "size": 19,
  },
  "annexText": "",
  "dateEnd": "",
  "dateRefusal": "",
  "dateRegistration": "31/10/2018",
  "dateStart": "13/03/2019",
  "deadline": "",
  "decisionUrl": "http://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=CELEX:ABACECD323&from=EN",
  "draftLegal": Object {
    "id": 16806,
    "mimeType": "text/plain",
    "name": "DummyMemberDoc1.txt",
    "size": 10,
  },
  "funding": Object {
    "totalAmount": 0,
  },
  "isPartiallyRegistered": undefined,
  "linguisticVersionIsFallback": false,
  "members": Array [
    Object {
      "fullName": "dfsadasdasd ASDASDASD",
      "privacyApplied": false,
      "type": "MEMBER",
    },
    Object {
      "fullName": "ewrwerwer EWRWER",
      "privacyApplied": false,
      "type": "MEMBER",
    },
    Object {
      "email": "asdasdqewrwe3@pepe.com",
      "fullName": "asdasdasdas ASDASDA",
      "privacyApplied": false,
      "type": "SUBSTITUTE",
    },
    Object {
      "fullName": "tyutyutyutyuty UTYUTY",
      "privacyApplied": false,
      "type": "MEMBER",
    },
    Object {
      "email": "eciinitiative1@gmail.com",
      "fullName": "ECI TEST",
      "privacyApplied": false,
      "type": "REPRESENTATIVE",
    },
    Object {
      "fullName": "qweqeqeqw QWEQWEQWE",
      "privacyApplied": false,
      "type": "MEMBER",
    },
    Object {
      "fullName": "asdsasda DASDAS",
      "privacyApplied": false,
      "type": "MEMBER",
    },
  ],
  "objectives": "End roaming fees across Europe now, ",
  "progress": Array [
    Object {
      "active": true,
      "date": "26/03/2019",
      "name": "ONGOING",
    },
    Object {
      "active": false,
      "date": "02/02/2019",
      "name": "ONGOING",
    },
    Object {
      "active": false,
      "date": "31/10/2018",
      "name": "REGISTERED",
    },
  ],
  "refusalReasons": Array [],
  "registrationNumber": "",
  "status": "ONGOING",
  "submission": Object {},
  "supportLink": "",
  "title": "DEVSingle Communication ",
  "treaties": "Article 114 ",
  "website": "https://webgate.ec.europa.eu/CITnet/confluence/display/",
}
`;
